src/main/java/adeuxpas/back/dto/mapper/AdMapper.java

```diff
@@ -0,0 +1,89 @@
+package adeuxpas.back.dto.mapper;
+
+import adeuxpas.back.dto.AdPostRequestDTO;
+import adeuxpas.back.dto.AdPostResponseDTO;
+
+import adeuxpas.back.entity.Ad;
+import adeuxpas.back.entity.ArticlePicture;
+
+import java.util.List;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.Named;
+
+/**
+ * Mapper interface for mapping Ad entities to DTOs (Data Transfer Objects).
+ * <p>
+ * This mapper provides methods for mapping Ad entities to various DTOs,
+ * such as its multiple article picture DTO
+ * </p>
+ * <p>
+ * It uses MapStruct annotations for mapping and Spring component model for integration with Spring.
+ * </p>
+ * <p>
+ * This interface defines mapping methods for converting Ad entity attributes to corresponding DTO fields.
+ * </p>
+ * <p>
+ * It also includes helper methods.
+ * </p>
+ * 
+ */
+@Mapper(componentModel = "spring")
+public interface AdMapper {
+    @Mapping(source = "articlePictures", target = "firstArticlePictureUrl", qualifiedByName = "findFirstArticlePictureUrl")
+    @Mapping(source = "articlePictures", target = "secondArticlePictureUrl", qualifiedByName = "findSecondArticlePictureUrl")
+    @Mapping(source = "articlePictures", target = "thirdArticlePictureUrl", qualifiedByName = "findThirdArticlePictureUrl")
+    @Mapping(source = "articlePictures", target = "fourthArticlePictureUrl", qualifiedByName = "findFourthArticlePictureUrl")
+    @Mapping(source = "articlePictures", target = "fifthArticlePictureUrl", qualifiedByName = "findFifthArticlePictureUrl")
+    @Mapping(source = "publisher.alias", target = "publisher")
+    @Mapping(source = "publisher.city", target = "publisherCity")
+    @Mapping(source = "publisher.postalCode", target = "publisherPostalCode")
+    AdPostResponseDTO adToAdResponseDTO(Ad ad);
+
```

Comment:
The name of this method should be 'adToAd**Post**ResponeDTO', otherwise it won't work

Reply:
it weirdly hasn't prevented it to work so far but I'll change it anyway thank you :)

Reply:
I think that's because you're not actually **using** the adMapper, you're doing the mappings manually in the service